src/merkle.rs

```diff
@@ -1810,6 +1810,37 @@ impl Element for [u8; 32] {
     }
 }
 
+pub fn get_merkle_tree_len_generic<
+    BaseTreeArity: Unsigned,
+    SubTreeArity: Unsigned,
+    TopTreeArity: Unsigned,
+>(
+    leaves: usize,
+) -> Result<usize> {
+    let top_tree_arity = TopTreeArity::to_usize();
+    let sub_tree_arity = SubTreeArity::to_usize();
+    let base_tree_arity = BaseTreeArity::to_usize();
+
+    let base_tree_len = match get_merkle_tree_len(leaves, base_tree_arity) {
```

Comment:
I think this can be written as `let base_tree_len = get_merkle_tree_len(leaves, base_tree_arity)?;`

Reply:
Definitely! Thanks!